datalad/utils.py

```diff
@@ -369,14 +369,25 @@ def _is_stream_tty(stream: Optional[IO]) -> bool:
 
 
 def is_interactive() -> bool:
-    """Return True if all in/outs are open and tty.
+    # TODO: Deprecate and refer to datalad.is_interactive instead.
+    #       Dependencies of datalad.__init__m datalad.config and
+    #       datalad.log on each other make the order of (partial) imports
+    #       critical.
+    from datalad import cfg
+    return cfg.obtain("datalad.ui.interactive")
 
-    Note that in a somewhat abnormal case where e.g. stdin is explicitly
-    closed, and any operation on it would raise a
-    `ValueError("I/O operation on closed file")` exception, this function
-    would just return False, since the session cannot be used interactively.
-    """
-    return all(_is_stream_tty(s) for s in (sys.stdin, sys.stdout, sys.stderr))
+
+def is_interactive_failsafe() -> bool:
```

Comment:
please add docstring to describe why we need this failsafe?

also again -- code/logic duplication.  Please e.g. rename original implementation into `_is_interactive` or `is_interactive_noconfig` or add `check_config=True` and overload for the config default?

the point is to not duplicate logic which might need fixing in multiple spots later.  also note that we have `@never_fail`  so may be of use directly within invoking in config default?

Reply:
Was unaware of never_fail indeed. Thx.

Reply:
Not sure, what duplication you are referring to, though.

Reply:
Just double-checked, but there's no duplication, I can see. `is_interactive()` now returns the config and the config gets its default from the new function, which is based on the former logic of `is_interactive`. It's a move of code, not duplication. What's the value of having that single line factored out again?

Reply:
I am sorry -- I think I have misread the diff, indeed no duplication, sorry about that noise.

Reply:
np